tools/clang/lib/CodeGen/CGHLSLMS.cpp

```diff
@@ -4267,34 +4267,15 @@ static bool CreateWriteEnabledStaticGlobals(llvm::Module *M) {
     }
   }
 
-  BasicBlock *mainFunctionEntryBlock = nullptr;
+  IRBuilder<> Builder(EF->getEntryBlock().getFirstNonPHI());
```

Comment:
You want first non-alloca; there can be no phis in the entry block.  Use dxilutil::FirstNonAllocaInsertionPt(llvm::BasicBlock* BB) with entry block instead.

Reply:
If there are no phis, it would just return the first instruction of the block. So it would be still valid.

Reply:
based on our offline discussion, I will switch to dxilutil::FirstNonAllocaInsertionPt as it is preferable to place instructions after alloca